Src/HALALMock/Services/Time/Time.cpp

```diff
@@ -1,445 +1,168 @@
-    Time::high_precision_alarms_by_id.erase(id);
-    for (auto timer : high_precision_timers) {
-        NVIC_EnableIRQ(timer32interrupt_by_timer32handler_map[timer]);
-    }
-
-    return true;
+    std::lock_guard<std::mutex> lock(mutex);
+    return alarms.erase(id) > 0;
 }
 
-uint8_t Time::register_mid_precision_alarm(uint32_t period_in_us,
-                                           function<void()> func) {
-    if (mid_precision_timer == nullptr) {
-        ErrorHandler(
-            "Cannot register mid precision alarm if no timer is used for mid "
-            "precision");
-        return 0;
-    }
-    if (mid_precision_alarms_by_id.size() ==
-        std::numeric_limits<uint8_t>::max()) {
-        ErrorHandler(
-            "Cannot register mid precision alarm as all id's are already "
-            "occupied");
-        return 0;
-    }
-    if (std::find_if(
-            TimerPeripheral::timers.begin(), TimerPeripheral::timers.end(),
-            [&](TimerPeripheral& tim) -> bool {
-                return tim.handle == mid_precision_timer && (tim.is_occupied());
-            }) != TimerPeripheral::timers.end()) {
-        ErrorHandler(
-            "a timer cannot be used as mid precision timer and PWM or Input "
-            "Capture Simultaneously");
-        return 0;  // TODO: put this check in high precision timers and global
-                   // timer too
-    }
-
-    Time::Alarm alarm = {
-        .period = period_in_us / Time::mid_precision_step_in_us,
-        .tim = Time::mid_precision_timer,
-        .alarm = [&]() {},
-        .offset = Time::mid_precision_tick,
-        .is_on = true};
-
-    while (mid_precision_alarms_by_id.contains(mid_precision_ids))
-        mid_precision_ids++;
-
-    NVIC_DisableIRQ(
-        timer32interrupt_by_timer32handler_map[mid_precision_timer]);
-    Time::mid_precision_alarms_by_id[mid_precision_ids] = alarm;
-
-    if (not Time::mid_precision_registered) {
-        hal_enable_timer(mid_precision_timer);
-        Time::init_timer(
-            timer32_by_timer32handler_map[mid_precision_timer],
-            Time::mid_precision_timer, 275, Time::mid_precision_step_in_us,
-            timer32interrupt_by_timer32handler_map[mid_precision_timer]);
-        Time::ConfigTimer(Time::mid_precision_timer,
-                          Time::mid_precision_step_in_us);
-        NVIC_DisableIRQ(
-            timer32interrupt_by_timer32handler_map[mid_precision_timer]);
-        Time::mid_precision_registered = true;
-    }
-
-    alarm.alarm = func;
-    Time::mid_precision_alarms_by_id[mid_precision_ids] = alarm;
-    NVIC_EnableIRQ(timer32interrupt_by_timer32handler_map[mid_precision_timer]);
-
-    return mid_precision_ids;
+uint8_t Time::register_low_precision_alarm(uint32_t period_in_ms, std::function<void()> func) {
+    return register_high_precision_alarm(period_in_ms * 1000, func); //There is no hw limitation that does not allow this
```

Comment:
you mean regarding Timers as they are currently configured in STLIB right?

Reply:
In HALAL there is three levels of timers because of hardware limitations, in the mock they do not exist, i just use high precision for everything and add extra methods to comply with the interface